### PR TITLE
Add and log condition id

### PR DESCRIPTION
**Pull request by DoctorVin**
#### What does this PR do
This PR expands on the use of the condition ID field to improve observability into a specific execution of a condition on a server

### CODE REVIEW

File: pkg/api/v1/routes/handlers.go [hunk 1]
```diff
@@ -270,10 +345,14 @@ func (r *Routes) serverConditionDelete(c *gin.Context) {
...
 			&v1types.ServerResponse{Message: "unsupported condition kind: " + string(kind)},
 		)
+		r.logger.WithFields(logrus.Fields{
+			"kind": kind,
+		}).Info("unsupported condition kind")
 
 		return
 	}
 
+	// TODO (vc): the repository Delete route should take a condition id
```
Comment (joelrebel) [inline pkg/api/v1/routes/handlers.go:355]: More of a discussion topic we can have separately, where do we persist the condition IDs.

File: pkg/api/v1/types/types.go [hunk 2]
```diff
@@ -59,31 +65,33 @@ type ConditionUpdate struct {
 //
 // This method makes sure that update does not overwrite existing data inadvertently.
 func (c *ConditionUpdate) MergeExisting(existing *ptypes.Condition) (*ptypes.Condition, error) {
+	// XXX: This make more sense for the caller to check?
```
Comment (joelrebel) [inline pkg/api/v1/types/types.go:68]: ah right I see what you mean't above, let's move this into the caller then.

**Review by joelrebel — CHANGES_REQUESTED** [review]
Looks good overall, left some comments - mostly just follow ups to your comments.

This would require a rebase after https://github.com/metal-toolbox/conditionorc/pull/5 is merged.

File: .golangci.yml [hunk 3]
```diff
@@ -73,7 +73,6 @@ linters:
     - noctx
     - stylecheck
     - whitespace
-    - wsl
```
[the remark this answers is not in the document]
Comment (DoctorVin) [reply .golangci.yml:76]: I should have asked before removing this. This got flagged on CI and after reading the linter and what it intends to enforce, I thought it achieved the exact opposite (specifically it aims to help readability and I think its enforcement of whitespace was not-at-all helpful). My opinion is that enforcing a given shape of the text on a screen is not very useful in terms of coding style.

I'm happy to add this back if you feel it adds value. I'd prefer we make it advisory, though.

File: pkg/api/v1/routes/handlers.go [hunk 4]
```diff
@@ -54,24 +61,31 @@ func (r *Routes) serverConditionUpdate(c *gin.Context) {
...
 			http.StatusBadRequest,
-			&v1types.ServerResponse{Message: "invalid ConditionUpdate payload, expected a valid resourceVersion"},
+			&v1types.ServerResponse{Message: "invalid ConditionUpdate payload: resourceVersion not set"},
 		)
+		r.logger.Info("resource version not set")
 
 		return
 	}
 
+	// XXX: shouldn't both of these fields be set in all cases? Even if status is something like
```
[the remark this answers is not in the document]
Comment (DoctorVin) [reply pkg/api/v1/routes/handlers.go:81]: I will add that in this PR. It's straightforward.

File: pkg/api/v1/types/types.go (hunk 2 above)
Comment (DoctorVin) [reply pkg/api/v1/types/types.go:68]: I'll update accordingly.

File: pkg/api/v1/routes/handlers.go [hunk 5]
```diff
@@ -80,23 +94,33 @@ func (r *Routes) serverConditionUpdate(c *gin.Context) {
...
+		r.logger.WithFields(logrus.Fields{
+			"serverID": serverID.String(),
+			"kind":     kind,
+		}).Info("condition not found")
 
 		return
 	}
 
 	// nothing to update
+	// XXX: consider just doing the update unconditionally?
```
[the remark this answers is not in the document]
Comment (DoctorVin) [reply pkg/api/v1/routes/handlers.go:123]: Let's think about it and revisit once we've gotten some runtime in the system.

File: .golangci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (joelrebel) [reply .golangci.yml:76]: no worries at all, like we discussed we can agree on which linters we want over time. Lets go with leaving this out and see if we end up missing it.

**Review by joelrebel — APPROVED** [review]
Thanks for the efforts here, LGTM!

File: pkg/api/v1/types/types.go (hunk 2 above)
Comment (DoctorVin) [reply pkg/api/v1/types/types.go]: I decided to pass on this. My rationale was that the ConditionUpdate member function can be called by any caller which may or may not use `nil` as parameter, so it's safer from that perspective to have duplicate checks.

**Comment by DoctorVin** [comment]
> Thanks for the efforts here, LGTM!

Sorry for the thrash, but I added a commit. Could you push the button again, please?

**Pull request merged**